docs/roleassumption.md

```diff
@@ -0,0 +1,393 @@
+account role assumption by using multiple profiles. 
+
+### Glossory
+
+* Target Account - The account where the cluster is created
+* Source Account - The AWS account where the CAPA controller runs. 
+
+## Assumptions
+1. The CAPA controllers are running in 1 AWS account and you want to create the target cluster in another AWS account.
+(We could also use this doc to create a cluster in the same account)
+2. This assumes that you start with no existing clusters. 
+
+## High level steps
+
+1. Creating a bootstrap/management cluster in AWS - This can be done by running the phases in clusterctl
+    * Uses the existing provider components yaml
+2. Setting up cross account roles
+3. Deploying the Kiam server/agent
+4. Create the target cluster (through kiam)
+    * Uses different provider components with no secrets and annotation to indicate the IAM Role to assume. 
+
+## 1. Creating the bootstrap cluster in AWS
+Using clusterctl command we can create a new cluster in AWS which in turn will act as the 
+bootstrap cluster to create the target cluster(in a different AWS account. This can be achieved by using the phases in 
+clusterctl to perform all the steps except the pivoting. This will provide us with a bare-bones functioning cluster that 
+we can use as a bootstrap cluster. 
+To begin with follow the steps in this getting started guide
+(https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/master/docs/getting-started.md) to setup the environment
+except for creating the actual cluster. Instead follow the steps below to create the cluster.
+
+create a new cluster using kind for bootstrapping purpose by running:
+```$xslt
+kind create cluster
+```
+and get its kube config path by running
+```
+export KIND_KUBECONFIG=`kind get kubeconfig-path`
+```
+
+Use the following commands to create new (bootstrap) cluster in AWS
+```
+kubectl alpha phases apply-cluster-api-components --provider-components cmd/clusterctl/examples/aws/out/provider-components.yaml \
+--kubeconfig $KIND_KUBECONFIG
+
+kubectl alpha phases apply-cluster --cluster cmd/clusterctl/examples/aws/out/cluster.yaml --kubeconfig $KIND_KUBECONFIG
+
+kubectl alpha phases apply-machines --machines cmd/clusterctl/examples/aws/out/machines.yaml 
+--kubeconfig $KIND_KUBECONFIG
+
+kubectl alpha phases get-kubeconfig --provider aws --cluster-name <CLUSTER_NAME> --kubeconfig $KIND_KUBECONFIG
+export AWS_KUBECONFIG=`pwd`/kubeconfig
+
+kubectl alpha phases apply-addons -a cmd/clusterctl/examples/aws/out/addons.yaml 
+--kubeconfig $AWS_KUBECONFIG
+
+```
+
+Verify that all the pods in the kube-system namespace are running smoothly. Also you may remove the additional node in 
+the machines example yaml since we are only interested in running the controllers that runs in control plane node
+(although its not required to make any changes there). You can destroy your local kind cluster by running 
+```$xslt
+make kind-reset
```

Comment:
this will delete the kind cluster named `clusterapi` but the above instructions create a kind cluster with the default name, `kind`